graylog2-server/src/main/java/org/graylog2/database/entities/source/EntitySource.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.database.entities.source;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.auto.value.AutoValue;
+import jakarta.annotation.Nullable;
+import org.graylog2.database.MongoEntity;
+import org.mongojack.Id;
+import org.mongojack.ObjectId;
+
+import java.util.Optional;
+
+@AutoValue
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonDeserialize(builder = EntitySource.Builder.class)
+public abstract class EntitySource implements MongoEntity {
+    // Source types
+    public static final String USER_DEFINED = "USER_DEFINED";
+    // Entity types
+    public static final String VIEW_TYPE = "view";
+
+    public static final String FIELD_ID = "id";
+    public static final String FIELD_SOURCE = "source";
+    public static final String FIELD_ENTITY_ID = "entity_id";
+    public static final String FIELD_ENTITY_TYPE = "entity_type";
+    public static final String FIELD_PARENT_ID = "parent_id";
+
+    @Nullable
+    @ObjectId
+    @JsonProperty(FIELD_ENTITY_ID)
+    public abstract String entityId();
+
+    @JsonProperty(FIELD_SOURCE)
+    public abstract String source();
+
+    // Since entity source objects can represent multiple types of entities, we specify the type only to simplify any
+    // potential future migrations. Outside of that use case, this field should not be referenced in application logic.
+    @JsonProperty(FIELD_ENTITY_TYPE)
+    public abstract String entityType();
```

Comment:
@ryan-carroll-graylog @danotorrey I know we discussed this field being collection name, so I wanted it to be required, but not returned to the user since they don't need to know that information. Without using a `@Nullable` annotation which opens it up to being left blank, there wasn't a good way I could find to do that between Jackson and AutoValue.

So I changed for an entity type that we can define here for open types and then in respective entity services in the enterprise repo. Let me know what you guys think.